common/src/main/java/com/wynntils/sockets/model/HadesModel.java

```diff
@@ -127,6 +135,22 @@ public void onPartyListUpdate(RelationsUpdateEvent.PartyList event) {
 
     @SubscribeEvent
     public void onWorldStateChange(WorldStateEvent event) {
+        if (event.getNewState() == WorldStateManager.State.WORLD && isSocketOpen() && firstWorldJoin) {
```

Comment:
How does this check for failure?

Reply:
isSocketOpen, socket is closed on auth fail.

Reply:
But this looks wrong? Don't you first need to do an if check if this is firstWorldJoin, and set the field, and then check socket? Also, the socket check looks inverted.